src/rgw/rgw_rados.cc

```diff
@@ -8579,6 +8579,14 @@ int RGWRados::cls_bucket_list_ordered(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+static int parse_index_hash_source(const string& oid_wo_ns, string *index_hash_source) {
```

Comment:
Please add a comment for the entire function that explains its narrow usage (removes everything from the second to last "." to the end) and that it's used to retrieve the hash source from an incomplete multipart entry.

Your use of namespace qualifiers is inconsistent. Please change `string` to `std::string`.

Your `if (...) return ...;`are against the ceph coding style (see _CodingStyle_ file at top level).